core/src/main/java/org/apache/calcite/sql/parser/StringAndPos.java

```diff
@@ -65,6 +101,10 @@ private StringAndPos(String sql, int cursor, @Nullable SqlParserPos pos) {
    * </ul>
    */
   public static StringAndPos of(String sql) {
+
+    if (BITXOR_OPERATOR_WHITELISTED_SQLS.contains(sql)) {
```

Comment:
the right way to do this is to introduce an escape mechanism, similar to escaping quotes in strings
For example, parse ^^ into a ^ in StringAndPos.
If you do that, please add test cases for the StringAndPos parser.